### PR TITLE
[DEV] update to v1.2.0rc2

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #67

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
I don't see any reason this won't have the same circular dependency problem as in #67 :-(

**Comment by xylar** [comment]
I'm including a patch for https://github.com/python-poetry/poetry/pull/5980.  Without that one, we can't make any headway here.

**Comment by xylar** [comment]
@maresb, any objection to me merging this including the patch to remove `poetry-plugin-export`?  I think this is a decent intermediate solution even though https://github.com/python-poetry/poetry/pull/5980 is still being debated.

**Comment by maresb** [comment]
I feel pretty uneasy about it. I commented about it some minutes ago, but I'm not finding it.

**Comment by maresb** [comment]
Ah, I wrote it in https://github.com/conda-forge/poetry-feedstock/issues/70#issuecomment-1231979495

**Comment by drhagen** [comment]
I am willing to test this build in various circumstances and report back, but I can't find instructions on how to install from a build. Is that possible?

**Comment by maresb** [comment]
@drhagen, it's a bit tricky. I just enabled an option for the GitHub Actions jobs to store the outputs as artifacts. Once it finishes, you should be able to download the package as a `.tar.bz2` and `mamba install` it.

Thanks! It's much appreciated.

File: conda-forge.yml
Comment (xylar) [inline conda-forge.yml]: @maresb, did you want this to be `True`?

Comment (maresb) [reply conda-forge.yml]: Oops!!! Ya.

But with the Cleo mess which you just pointed out, it looks like 1.2 is not going to happen.

I would hope that:
* most projects need only `poetry-core` in their host dependencies, and
* the poetry devs stabilize their release schedule in a reasonable timeframe

If those two work out, maybe this isn't a total disaster.

**Comment by xylar** [comment]
@maresb, can you now provide instructions on downloading the build artifacts?  I couldn't figure it out.

**Comment by maresb** [comment]
Ah, sorry, I forgot to rerender!

**Comment by maresb** [comment]
@conda-forge-admin, please rerender

**Comment by maresb** [comment]
If you open the logs there should be an obvious option to download artifacts, when the artifacts actually build.

**Comment by maresb** [comment]
See "1 artifact produced":
image

**Comment by xylar** [comment]
@maresb  I'm not sure how much it will help with #72 but what do you see as the downside of merging this PR (1.2.0rc2) with the patch to remove `poetry-plugin-export`? I agree that merging #72 with that patch is probably too risky but it seems like few folks are using the dev version and those who are won't be surprised if it's a little experimental.  It would probably help @drhagen to be able to play around with it.

**Comment by maresb** [comment]
Ah, right, that makes sense. I was getting confused between the `main` release branch and the `dev` prerelease branch. I agree that rc2 is experimental, so we can break. Let's merge!

**Review by maresb — APPROVED** [review]

**Comment by drhagen** [comment]
Installing this release was quite the trip, but in the end was possible with this Mambaforge command:

```
conda install --channel "conda-forge/label/poetry_dev" --channel "conda-forge/label/cleo_dev" poetry=1.2.0rc2
```

I tested the following Poetry commands and everything seems to be working fine:

- `poetry install` from an existing lock file
- `poetry lock; poetry install` from an existing lock file
- `poetry install` in a repo with no poetry.lock file
- `poetry lock; poetry install` in a repo with no poetry.lock file
- `poetry run nox` in a project with Nox

I did these tests both in the `base` environment as well as in a project-specific environment, because [Poetry behaves differently when it is in a Conda environment](https://github.com/python-poetry/poetry/blob/65ff74a9253fa2ea7f1169e8d9f2cb91a5fd0ab8/src/poetry/utils/env.py#L677-L684).

The base environment worked exactly as expected.

Strangely, Poetry seems to ignore the conda environment and make its own environment. It does not look like this code changed in 1.2.0, and it does the same thing if I install rc2 from pip or install 1.1.14. This either broke a long time ago or I am doing something wrong.

Also, if I do `poetry export`, I get a `The command "export" does not exist.` error as one would expect given the patch.

**Comment by maresb** [comment]
> Installing this release was quite the trip

Ya, I wrote instructions in https://github.com/conda-forge/poetry-feedstock/issues/64

> Strangely, Poetry seems to ignore the conda environment and make its own environment.

I work mostly Dockerized, and I usually run

```bash
poetry config virtualenvs.create false
```

which prevents virtualenv creation.

**Comment by xylar** [comment]
@drhagen, thanks for this testing!

Sorry the installation gave you trouble but it's true that it's tricky to install dev packages, particularly like here with dev dependencies.

> Also, if I do `poetry export`, I get a `The command "export" does not exist.` error as one would expect given the patch.

Is that no longer true if you manually install `poetry-plugin-export`?

**Comment by xylar** [comment]
To answer my own question, it seems to work.  I will work on getting https://github.com/conda-forge/poetry-plugin-export-feedstock/pull/4 to build so we have the "right" version of `poetry-plugin-export` to install alongside `poetry` 1.2.0rc2.

**Comment by xylar** [comment]
@maresb, how are you feeling at this point about moving forward with a patched 1.2.0?  Still feeling like it's better not to have 1.2.0 at all?

**Comment by xylar** [comment]
I guess the answer is that we're still blocked by not having a compatible release of `cleo`, so no worries for now.

**Comment by maresb** [comment]
In general I feel very reluctant to patch non-Conda-specific functionality in the packages we're building. I worry that it might cause more confusion than it solves, but that's just my opinion.

If the cleo issue clears up, perhaps it'd be worthwhile to seek another opinion from someone wiser and more experienced than me, like ocefpaf.

**Comment by xylar** [comment]
I understand that.  I see this case as being a little different because pip/pypi support circular dependencies whereas conda doesn't and we're trying to get around it.  Even so, there's certainly a high risk of confusion or unintended consequences.

I worry that there will be a great deal of confusion and frustration if we just say folks can't use poetry 1.2.0.  But we can see.

We can certainly ask ocefpaf when the time comes.  I think he would agree with my assessment that he's not necessarily the most cautious person when it comes to this kind of thing.  I think he'd be okay with patching and dealing with the consequences.  But we can certainly ask.

**Comment by maresb** [comment]
>I think he's be okay with patching and dealing with the consequences. 

It's more than I'd personally want to take on, but since you seem motivated I certainly don't want to stop you. :smile:

**Comment by drhagen** [comment]
> > Also, if I do `poetry export`, I get a `The command "export" does not exist.` error as one would expect given the patch.
> 
> Is that no longer true if you manually install `poetry-plugin-export`?

`poetry export` works fine if I conda install `poetry-plugin-export`.

**Pull request merged**
